features/netsocket/NetworkStack.h

```diff
@@ -37,13 +39,10 @@ class NetworkStack
     virtual ~NetworkStack() {};
 
     /** Get the local IP address
-     *  @deprecated
      *
      *  @return         Null-terminated representation of the local IP address
      *                  or null if not yet connected
      */
-    MBED_DEPRECATED_SINCE("mbed-os-5.7",
-        "Use NetworkInterface::get_ip_address()")
     virtual const char *get_ip_address();
```

Comment:
Why is this being undeprecated?

Reply:
It actually never has been - that's a proposed deprecation on the feature-emac branch.

On reflection we've decided this is too useful as "get primary network address for this stack", which is what the DNS code among other wants to do to get hints for IPv6-ness, so we've decided not to deprecate.

Note that DNS is definitely a stack thing, but there's an issue that we've no way to get from stack to interface(s). If we were to add `NetworkInterface *NetworkStack::get_default_interface()` or `NetworkInterface *NetworkStack::get_interface(int)`, we would be able to use those, rather than have these funny network stack calls that implicitly map to default interface.

Reply:
Ah I see :+1: